test/log-iterator.spec.js

```diff
@@ -0,0 +1,243 @@
+
+        // lte + gt
+        let it_lte_gt = log1.iterator({
+          gt: 'zdpuApQ4crttH3GmnVQPNdZ5UMNv3aCd3bcccFEKYCAdYUzir',
+          lte: 'zdpuAxdV4PTwMZ4Ug2r9Y1Vi48EUJW4YZPnaWZW9kSfbR82SF'
+        })
+        let it_lte_gt_array = [...it_lte_gt]
+
+        // neither hash should appear in the array
+        assert.strictEqual(it_lte_gt_array.map(e => e.hash)
+          .indexOf('zdpuApQ4crttH3GmnVQPNdZ5UMNv3aCd3bcccFEKYCAdYUzir'), -1)
+        assert.strictEqual(it_lte_gt_array.map(e => e.hash)
+          .indexOf('zdpuAxdV4PTwMZ4Ug2r9Y1Vi48EUJW4YZPnaWZW9kSfbR82SF'), 0)
+        assert.strictEqual(it_lte_gt_array.length, 4)
+
+        // lte + gte
+        let it_lte_gte = log1.iterator({
+          gte: 'zdpuAmAvVqqHiss5arZoY6oNDN6JRLVExfWyN48rxTrsscHaE',
+          lte: 'zdpuApYv1r3kPGdSJoDYA8xkEAzKFK3bdvrdRX4jY13xZDpL6'
+        })
+        let it_lte_gte_array = [...it_lte_gte]
+
+        // neither hash should appear in the array
+        assert.strictEqual(it_lte_gte_array.map(e => e.hash)
+          .indexOf('zdpuAmAvVqqHiss5arZoY6oNDN6JRLVExfWyN48rxTrsscHaE'), 9)
+        assert.strictEqual(it_lte_gte_array.map(e => e.hash)
+          .indexOf('zdpuApYv1r3kPGdSJoDYA8xkEAzKFK3bdvrdRX4jY13xZDpL6'), 0)
+        assert.strictEqual(it_lte_gte_array.length, 10)
+      })
+      /* eslint-enable camelcase */
+
+      it('iterates with gt and default amount value (full log)', async () => {
+        let it = log1.iterator({
+          gt: 'zdpuAx5FC3cgoQSW7oXpBw6x4YiuDuaF9BBgncguzvndP8tDL'
+        })
+
+        let i = 0
+        let count = 0
+        for (let entry of it) {
+          assert.strictEqual(entry.payload, 'entry' + (100 - i++))
+          count++
+        }
+        assert.strictEqual(count, 33)
+      })
+
+      it('iterates with gte and default amount value (full log)', async () => {
+        let it = log1.iterator({
+          gte: 'zdpuAx5FC3cgoQSW7oXpBw6x4YiuDuaF9BBgncguzvndP8tDL'
+        })
+
+        let i = 0
+        let count = 0
+        for (let entry of it) {
+          assert.strictEqual(entry.payload, 'entry' + (100 - i++))
+          count++
+        }
+        assert.strictEqual(count, 34)
+      })
+
+      it('iterates with lt and default amount value (full log)', async () => {
+        let it = log1.iterator({
+          lt: 'zdpuAx5FC3cgoQSW7oXpBw6x4YiuDuaF9BBgncguzvndP8tDL'
+        })
+
+        assert.strictEqual([...it].length, 67)
+
+        let i = 0
+        for (let entry of it) {
+          assert.strictEqual(entry.payload, 'entry' + (67 - i++))
+        }
+      })
+
+      it('iterates with lte and default amount value (full log)', async () => {
+        let it = log1.iterator({
+          lte: 'zdpuAx5FC3cgoQSW7oXpBw6x4YiuDuaF9BBgncguzvndP8tDL'
+        })
+
+        assert.strictEqual([...it].length, 68)
+
+        let i = 0
+        for (let entry of it) {
+          assert.strictEqual(entry.payload, 'entry' + (67 - i++))
+        }
```

Comment:
The check for the right order as done in these loops could be separated to their own tests.